lib/Fhp/Response/Response.php

```diff
@@ -328,7 +328,11 @@ protected function findSegments($name, $one = false)
      */
     protected function splitSegment($segment)
     {
-        $parts = explode('+', $segment);
+        $parts = preg_split('/\+(?<!\?\+)/', $segment);
+
+        foreach ( $parts as &$part ) {
+            $part = str_replace( '?+', '+', $part );
```

Comment:
Please change 
`str_replace( '?+', '+', $part );`
to
`str_replace('?+', '+', $part);`